File: Data Structures & Algorithms/two-integer-sum/submission-6.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        const seen = new Map();

        for (let i = 0; i < nums.length; i++) {
            let goal = target - nums[i];
            if (seen.has(goal)) {
                return [seen.get(goal), i];
            }
            seen.set(nums[i], i);
        }

        return [];
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-6.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        seen = set()
        res = 0

        for num in nums:
            seen.add(num)

        for num in seen:
            if (num - 1) not in seen:
                length = 1
                while num+length in seen:
                    length += 1
                res = max(res, length)

        return res